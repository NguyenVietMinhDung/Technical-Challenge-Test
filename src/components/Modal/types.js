export type Props = {
  isVisible: boolean,
  toggleModal: Function,
};
