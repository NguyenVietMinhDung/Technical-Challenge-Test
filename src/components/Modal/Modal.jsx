import React from 'react';
import CSSModules from 'react-css-modules';
import styles from './Modal.scss';
import type { Props } from './types';

const Modal = (props: Props) => {
  const { selectedImgUrl, isVisible, toggleModal } = props;
  const handleClickEvent = () => toggleModal(null);
  return (
    isVisible ? (
      <>
        <div
          styleName="black-mask"
          onClick={handleClickEvent}
        />
        <div styleName="container">
          <img
            styleName="modal-body"
            src={selectedImgUrl}
            alt=""
          />
        </div>
      </>
    ) : null
  );
};

export default CSSModules(Modal, styles);