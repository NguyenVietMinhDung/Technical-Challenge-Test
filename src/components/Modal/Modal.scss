@import '../../styles/variables';

.black-mask {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: $black-mask-color;
}

.container {
  position: fixed;
  top: 25px;
  left: 250px;
  width: calc(100% - 500px);
  height: calc(100% - 50px);
  background-color: $white-color;
  box-sizing: border-box;
  padding: 10px;

  .modal {
    &-body {
      width: 100%;
      height: 100%;
    }
  }

  @media (max-width: $tablet-max-width) {
    top: 200px;
    left: 50px;
    width: calc(100% - 100px);
    height: calc(100% - 400px);
  }

  @media (max-width: $mobile-max-width) {
    top: 200px;
    left: 25px;
    width: calc(100% - 50px);
    height: calc(100% - 400px);
  }
}