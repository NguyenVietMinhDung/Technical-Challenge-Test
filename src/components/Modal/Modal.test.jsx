import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import gifImgUrl from '../../assets/giphy.gif';
import Modal from './Modal';

const defaultProps = {
  selectedImgUrl: gifImgUrl,
  isVisible: true,
  toggleModal: jest.fn(),
};

describe('<Modal />', () => {
  test('renders correctly', () => {
    const component = shallow(<Modal {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  test('handle click event', () => {
    const component = mount(<Modal {...defaultProps} />);
    component.find('div:first-child').simulate('click');
    component.setProps({ isVisible: false, selectedImgUrl: null });
    expect(component.props().toggleModal).toBeCalledWith(null);
  });
});
