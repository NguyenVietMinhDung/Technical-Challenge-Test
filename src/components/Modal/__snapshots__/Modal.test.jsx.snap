// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
<Fragment>
  <div
    className="black-mask"
    onClick={[Function]}
  />
  <div
    className="container"
  >
    <img
      alt=""
      className="modal-body"
      src="test-file-stub"
    />
  </div>
</Fragment>
`;
